examples/persistent_state.rs

```diff
@@ -0,0 +1,79 @@
+extern crate libsystemd;
+
+use libsystemd::activation;
+use libsystemd::daemon::{self, NotifyState};
+use std::io::{Read, Seek, Write};
+use std::os::unix::prelude::{AsRawFd, IntoRawFd};
+
+use memfd::Memfd;
+
+/*
+cargo build --example persistent_state
+systemd-run --user -p RestartForceExitStatus=2 -p SuccessExitStatus=2 -p FileDescriptorStoreMax=1 --wait ./target/debug/examples/persistent_state
```

Comment:
This is lovely :heart_eyes:
I'd move these instructions to a top-level docstring and preface it with a more verbose description of the counting game which this example is playing.